### ISSUE TITLE
Characters are highlighted misleadingly in vi-mode visual selection

### ISSUE THREAD

**Issue by doronbehar** [zsh-users/zsh-syntax-highlighting#637]
I'm experiencing an annoyance using `bindkey -v` and visual mode selection only when this plugin is enabled. I'll explain:

Say my $BUFFER has the following text, current $KEYMAP is "vicmd", and
the cursor is located where ^ is:

```
ZSH is the best shell
              ^
```

I press `vF<space>` in order to highlight the characters marked with `-----`:

```
ZSH is the best shell
          ^
          -----            (selected characters)
```
Depending on the contents of the `$BUFFER`, sometimes the letter `t` of the word `best` isn't highlighted, for example:

If my `$BUFFER` has:

```
git commit -m "ZSH is the best shell"
```

The string `ZSH is the best shell` will be colored (`fg=yellow` with my config) and performing the same actions will result in the letter `t` not highlighted.

In comparison, if the region selected isn't affected by z-sy-h, or the plugin is completely disabled, the letter `t` **is highlighted**.

**Comment by danielshahaf** [comment]
Works for me, xterm 327-2, xfce4-terminal 0.8.3-1, zsh HEAD, z-sy-h HEAD.

Could you please show the output of `typeset -p ZSH_VERSION ZSH_PATCHLEVEL ZSH_HIGHLIGHT_REVISION ZSH_HIGHLIGHT_VERSION`? What's the name and version of your terminal emulator?

And humour me, but could we please use [neutral text](https://en.wikipedia.org/wiki/Lorem_ipsum) for bug reports?

**Comment by danielshahaf** [comment]
Can you reproduce this in `zsh -f` + `source zsh-syntax-highlighting.zsh`?

**Comment by doronbehar** [comment]
Daniel here's a demonstration of the behavior using a desktop recording:

![z-sy-h-nohighlight-issue](https://user-images.githubusercontent.com/10998835/66747939-5966bb80-ee75-11e9-931e-4c278ce3537f.gif)

As you can see, the letter `t` isn't highlighted when the cursor is at the other end of the selection, only when this part of the buffer is already handled by z-sy-h. Otherwise, `t` _is_ highlighted.

I'm using urxvt and here's my [`.Xresources`](https://github.com/doronbehar/dotfiles/blob/5d9d8ca/.Xresources#L103).

Here's my urxvt version:

```.
rxvt-unicode (urxvt) v9.22 - released: 2016-01-23
options: perl,xft,styles,combining,blink,iso14755,unicode3,encodings=eu+vn+jp+jp-ext+kr+zh+zh-ext,fade,transparent,tint,pixbuf,XIM,frills,selectionscrolling,wheel,slipwheel,cursorBlink,pointerBlank,scrollbars=plain+rxvt+NeXT+xterm
Usage: urxvt [-help] [--help]
 [-display string] [-tn string] [-geometry geometry] [-C] [-iconic]
 [-cd string] [-dockapp] [-/+rv] [-/+ls] [-mc number] [-/+j] [-/+ss] [-/+ptab]
 [-/+sb] [-/+sr] [-/+st] [-sbt number] [-/+si] [-/+sk] [-/+sw] [-/+ip] [-/+tr]
 [-tint color] [-sh number] [-blr HxV] [-fade number] [-fadecolor color]
 [-/+ut] [-/+vb] [-/+tcw] [-/+insecure] [-/+uc] [-/+bc] [-/+pb] [-bg color]
 [-fg color] [-hc color] [-cr color] [-pr color] [-pr2 color] [-bd color]
 [-pixmap file[;geom]] [-icon file] [-fn fontname] [-fb fontname]
 [-fi fontname] [-fbi fontname] [-/+is] [-im name] [-pt style]
 [-imlocale string] [-imfont fontname] [-name string] [-title string]
 [-n string] [-sl number] [-embed windowid] [-depth number] [-visual number]
 [-/+override-redirect] [-pty-fd fileno] [-/+hold] [-w number] [-b number]
 [-/+bl] [-lsp number] [-letsp number] [-/+sbg] [-mod modifier] [-/+ssc]
 [-/+ssr] [-pe string] [-e command arg ...]
```

The same behavior is experienced with Xterm. My  `.Xresources` controls it's settings as well.

```
$ xterm -version
XTerm(349)
```

Here's more info:

```
$ typeset -p ZSH_VERSION ZSH_PATCHLEVEL ZSH_HIGHLIGHT_REVISION ZSH_HIGHLIGHT_VERSION
typeset ZSH_VERSION=5.7.1
typeset ZSH_PATCHLEVEL=zsh-5.7.1-0-g8b89d0d
typeset ZSH_HIGHLIGHT_REVISION=HEAD
typeset ZSH_HIGHLIGHT_VERSION=0.7.0-dev
```

**Comment by danielshahaf** [comment]
Fixed in #638, thanks Oliver.

**Comment by doronbehar** [comment]
Awesome, it works. Thanks!
